### PR TITLE
Add hashed column for API keys

### PR DESCRIPTION
**Pull request by tommoor**
Towards #7675 – adds news column for storing the API key as a hashed value, and a preview of that value.

### CODE REVIEW

File: app/scenes/Settings/components/ApiKeyListItem.tsx [hunk 1]
```diff
@@ -52,11 +52,21 @@ const ApiKeyListItem = ({ apiKey, isCopied, onCopy }: Props) => {
...
+              </Button>
+            </CopyToClipboard>
+          )}
+          <Text
+            type="tertiary"
+            size="small"
+            style={{ marginRight: 8 }}
+            monospace
+          >
+            ol...{apiKey.last4}
```
Comment (tommoor) [inline app/scenes/Settings/components/ApiKeyListItem.tsx:68]: As the API key is no longer available after creation, we show a preview of the value instead to allow identification.

File: server/migrations/20240929194201-add-hash-to-api-key.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  async down (queryInterface, Sequelize) {
+    await queryInterface.sequelize.transaction(async transaction => {
+      await queryInterface.removeColumn("apiKeys", "hash", { transaction });
+      await queryInterface.removeColumn("apiKeys", "last4", { transaction });
+      await queryInterface.changeColumn("apiKeys", "secret", {
+        type: Sequelize.STRING,
+        allowNull: false,
+      }, { transaction });
+    });
```
Comment (apoorv-mishra) [inline server/migrations/20240929194201-add-hash-to-api-key.js:33]: Note that this would fail because `secret` would have `null` values!

Comment (tommoor) [reply server/migrations/20240929194201-add-hash-to-api-key.js:33]: If any new API keys have been created it does fail yea, they need removing from the DB before rolling back.

Unfortunately by definition this change is lossy and non-revertable.

**Review by apoorv-mishra — APPROVED** [review]
LGTM 👍

**Pull request merged**